### PR TITLE
Allows use of numeric properties

### PR DESCRIPTION
**Pull request by bkadatz**
The [Serilog wiki](https://github.com/serilog/serilog/wiki/Writing-Log-Events#message-template-syntax) states:

> - Formats that use numeric property names, like {0} and {1} exclusively, will be matched with the log method's parameters by treating the property names as indexes; this is identical to string.Format()'s behaviour
> - If any of the property names are non-numeric, then all property names will be matched from left-to-right with the log method's parameters

However, when calling `log.Information("Hello {0}", name);` this causes an Azure storage exception as columns in Azure Table Storage must follow [naming rules for C# identifiers](https://blogs.msdn.microsoft.com/jmstall/2014/06/12/azure-storage-naming-rules/) and the name `0` is invalid.

This fix updates the code so that if it detects a numeric property name, it won't attempt to add that as a separate property to table storage. However, for completeness the property is added to the `AggregatedProperties` table property. It also supports mixed use of numeric and named properties.

### CODE REVIEW

File: src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableStorageWithProperties/AzureTableStorageEntityFactory.cs [hunk 1]
```diff
@@ -61,9 +61,15 @@ public static DynamicTableEntity CreateEntityWithProperties(LogEvent logEvent, I
 
             List<KeyValuePair<ScalarValue, LogEventPropertyValue>> additionalData = null;
             var count = dynamicProperties.Count;
+			bool isNumeric;
```
Comment (nblumhardt) [inline src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableStorageWithProperties/AzureTableStorageEntityFactory.cs]: Nit: indent seems problematic here.

File: src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableStorageWithProperties/AzureTableStorageEntityFactory.cs [hunk 2]
```diff
@@ -61,9 +61,15 @@ public static DynamicTableEntity CreateEntityWithProperties(LogEvent logEvent, I
...
+			int parsedNumber;
+
             foreach (var logProperty in logEvent.Properties)
             {
-                if (count++ < _maxNumberOfPropertiesPerRow - 1)
+				isNumeric = int.TryParse(logProperty.Key, out parsedNumber);
```
Comment (nblumhardt) [inline src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableStorageWithProperties/AzureTableStorageEntityFactory.cs]: Can we encapsulate this check in a function `IsValidColumnName(s)`? Although the check for numerics is an improvement, Serilog allows (e.g. through enrichers) all kinds of property names, for example `hello-world` or `a.b:c`. In the future we'll need to handle these cases properly, too.

**Comment by nblumhardt** [comment]
LGTM, minor comments.

Thanks for taking this one on - looks like an important change.

File: src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableStorageWithProperties/AzureTableStorageEntityFactory.cs (hunk 1 above)
Comment (bkadatz) [inline src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableStorageWithProperties/AzureTableStorageEntityFactory.cs]: Looks like a tab vs space issue. I'll clean that up.

File: src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableStorageWithProperties/AzureTableStorageEntityFactory.cs (hunk 2 above)
Comment (bkadatz) [inline src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableStorageWithProperties/AzureTableStorageEntityFactory.cs]: Considering this use case a bit more, it may be useful to treat numeric properties separately from generally invalid property names. If someone uses `{0}`, `{1}`, etc. then they're obviously not wanting these to be stored as a separate property in the table (otherwise they'd have specified a name) and likely not in the `AggregateProperties` table property either, as long as they are in the rendered message. If it's not a numeric property name, then a separate check to ensure it conforms to C# identifier naming conventions makes sense. I'll add that function and validate the name using a C# identifier regex.

In short, numeric properties should probably be skipped. All other properties are either stored in a table property if the name is valid or, if the name isn't valid, in the `AggregateProperties` table property.

Comment (nblumhardt) [inline src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableStorageWithProperties/AzureTableStorageEntityFactory.cs]: That's a reasonable assumption - though sometimes numeric property names provide a "free" upgrade from text -> structured logs, so other sinks treat them as normal.
